test_infra/src/lib.rs

```diff
@@ -844,10 +846,20 @@ pub const PIPE_SIZE: i32 = 32 << 20;
 
 static NEXT_VM_ID: LazyLock<Mutex<u8>> = LazyLock::new(|| Mutex::new(1));
 
+#[repr(i32)]
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+pub enum GuestParamKey {
+    FwPath = 0,
+    KernelPath = 1,
+    KernelCmdline = 2,
+    BootTimeout = 3,
+}
 pub struct Guest {
     pub tmp_dir: TempDir,
     pub disk_config: Box<dyn DiskConfig>,
     pub network: GuestNetworkConfig,
+    pub vm_type: GuestVmType,
+    pub params: HashMap<GuestParamKey, Box<dyn Any>>,
```

Comment:
This approach seems unnecessarily complex. Could we simplify by adding new fields to the `Guest` struct instead of introducing a `HashMap`?